### ISSUE TITLE
Allow redirectTo to take a regular expression as an argument

### ISSUE THREAD

**Issue by mikemfleming** [chaijs/chai-http#235]
Right now you can only use this method to test a redirect to a static destination string.

If you were testing that an article redirected to something like `/widgets/:widget_id` you couldn't use a direct comparison to a string like it's currently doing.

Request: configure it to accept either a string or a regex argument. Perform the check on items in the `redirects` array using either the string or the regex. 

https://github.com/chaijs/chai-http/blob/d5156913d72f4bbfacdd9899c65f60ecaf397892/lib/http.js#L288

**Comment by austince** [comment]
Hey @mikemfleming, sorry for the delay but this seems like a good improvement to me. You've found the right file to update. I believe we would just have to update this:

https://github.com/chaijs/chai-http/blob/d5156913d72f4bbfacdd9899c65f60ecaf397892/lib/http.js#L286-L292

to something like:

```js
    if (redirects && redirects.length) {
      var hasRedirected; // could definitely be cleaned up, but the gist
      if (Object.prototype.toString.call(destination) == '[object RegExp]') {
        hasRedirected = false;
        for (int i = 0; i < redirects.length; i++) {
           hasRedirected = destination.test(redirects[i]);
           if (hasRedirect) { 
             break;
          }
        }
      } else {  
        hasRedirected = redirects.indexOf(destination) > -1;
     }

      this.assert( 
         hasRedirected,
         , 'expected redirect to ' + destination + ' but got ' + redirects.join(' then ') 
         , 'expected not to redirect to ' + destination + ' but got ' + redirects.join(' then ') 
       ); 
    }
```

Is this a feature you would like to try your hand at implementing?

**Comment by mikemfleming** [comment]
@austince That sounds good to me, I'll start working on it this weekend.

**Comment by austince** [comment]
Closed by https://github.com/chaijs/chai-http/pull/243 :confetti_ball: - nice work @mikemfleming !

**Comment by austince** [comment]
Released in [4.3.0](https://github.com/chaijs/chai-http/releases/tag/4.3.0) :balloon:
